### PR TITLE
[AIRFLOW-6220] BigQuery - remove redundant hook tests after renaming

### PR DESCRIPTION
**Pull request by TobKed**
These tests were removed in https://github.com/apache/airflow/pull/6729 but came by mistake in https://github.com/apache/airflow/pull/6734.

Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-6220
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

File: tests/gcp/hooks/test_bigquery_system.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ def setUp(self):
         self.instance = hook.BigQueryHook()
 
     @mock.patch(
```
Comment (mik-laj) [inline tests/gcp/hooks/test_bigquery_system.py]: Why do we do mocking in system tests?

Comment (TobKed) [reply tests/gcp/hooks/test_bigquery_system.py]: Excellent point! I've verified are tests working without and removed them.

File: tests/gcp/hooks/test_bigquery_system.py [hunk 2]
```diff
@@ -31,51 +30,26 @@ class BigQueryDataframeResultsSystemTest(unittest.TestCase):
     def setUp(self):
         self.instance = hook.BigQueryHook()
 
-    @mock.patch(
-        'airflow.gcp.hooks.base.GoogleCloudBaseHook.project_id',
-        new_callable=mock.PropertyMock,
-        return_value=None
-    )
-    def test_output_is_dataframe_with_valid_query(self, mock_project_id):
+    def test_output_is_dataframe_with_valid_query(self):
```
Comment (mik-laj) [inline tests/gcp/hooks/test_bigquery_system.py]: Is it related to this PR?

Comment (TobKed) [reply tests/gcp/hooks/test_bigquery_system.py]: Actually it is not. I will rollback these changes and create separate PR for this change.

File: tests/gcp/hooks/test_bigquery_system.py (hunk 1 above)
Comment (mik-laj) [reply tests/gcp/hooks/test_bigquery_system.py]: Any progress on it? I would like to accept these changes.

**Pull request merged**